tests/dagger/stores/testblockexc.nim

```diff
@@ -118,58 +134,63 @@ suite "BlockExc engine - 2 nodes":
       sendDontHave: false)
 
     peerCtx1.peerWants.add(entry)
-    check blockexc2.taskQueue.pushOrUpdateNoWait(peerCtx1).isOk
+    check blockexc2.engine.taskQueue.pushOrUpdateNoWait(peerCtx1).isOk
     await sleepAsync(100.millis)
 
     check blockexc1.engine.localStore.hasBlock(blk.cid)
 
   test "should get blocks from remote":
-    let blocks = await blockexc1.getBlocks(blocks2.mapIt( it.cid ))
-    check blocks == blocks2
+    let blocks = await allFinished(
+      blocks2.mapIt( blockexc1.getBlock(it.cid) ))
+    check blocks.mapIt( !it.read ) == blocks2
 
   test "remote should send blocks when available":
-    let blk = !bt.Block.new("Block 1".toBytes)
+    let blk = bt.Block.new("Block 1".toBytes)
 
     # should fail retrieving block from remote
-    check not await blockexc1.getBlocks(@[blk.cid])
+    check not await blockexc1.getBlock(blk.cid)
       .withTimeout(100.millis) # should expire
 
-    proc onBlocks(evt: BlockStoreChangeEvt) =
-      check evt.cids == @[blk.cid]
-      done.complete()
-
-    blockexc1.engine.localStore.addChangeHandler(onBlocks, ChangeType.Added)
-
     # first put the required block in the local store
-    blockexc2.engine.localStore.putBlocks(@[blk])
+    await blockexc2.engine.localStore.putBlock(blk)
 
     # second trigger blockexc to resolve any pending requests
     # for the block
-    blockexc2.putBlocks(@[blk])
+    await blockexc2.putBlock(blk)
 
-    await done
+    # should fail retrieving block from remote
+    check await blockexc1.getBlock(blk.cid)
+      .withTimeout(100.millis) # should succede
```

Comment:
It's likely related to working my way through a "moving target" (I realize you're actively working on this) but comments on L161 and L163 seem to be at odds, or have I misunderstood something?

Reply:
Ah, good catch, the comment should be changed. The idea is to first make a request for a non existent block and verify that it fails which is done on line 151, then we add the block to the store and request it again and this time it should actually pass.